hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmMultipartKeyInfo.java

```diff
@@ -109,15 +189,18 @@ public long getCreationTime() {
   }
 
   public TreeMap<Integer, PartKeyInfo> getPartKeyInfoMap() {
-    return partKeyInfoList;
+    TreeMap<Integer, PartKeyInfo> treeMap = new TreeMap<>();
+    partKeyInfoMap.forEach(
+        partKeyInfo -> treeMap.put(partKeyInfo.getPartNumber(), partKeyInfo));
```

Comment:
@xichen01 , it should not be copied to a TreeMap.  It takes time and memory.

Reply:
The whole multiupload process will only call `getPartKeyInfoMap` once even if a lot of key parts are uploaded, so this `TreeMap` doesn't consume a lot of memory, if we don't return `TreeMap`, we need to change a lot of existing code.

![image](https://github.com/apache/ozone/assets/32928346/517bab4c-fd19-477b-8a84-ba04a103a799)
`getPartKeyInfoMap` will not call when `S3MultipartUploadCommitPartRequest`

Reply:
> The whole multiupload process will only call getPartKeyInfoMap once (in the S3MultipartUploadCompleteRequest)  ...

@xichen01 , This is the behavior today.  Someone may call getPartKeyInfoMap and misuse it in the future.  Please update the corresponding code.